src/lib/scheduler_clients/slurm/slurm_client.py

```diff
@@ -54,9 +55,15 @@ async def submit_job(
     ) -> int | None:
 
         if job_description.script_path:
-            return await self.slurm_cli_client.submit_job(
-                job_description, username, jwt_token
-            )
+            if self.ssh_client:
```

Comment:
I recommend additional refactoring.

Initialize SlurmCliClient only if ssh_client is available
```
self.slurm_cli_client = None
if ssh_client:
   self.slurm_cli_client = SlurmCliClient(ssh_client, slurm_version)

```

In submit_job check directly the slurm_cli_client instead of ssh_client
```
if self. slurm_cli_client:
   return await self.slurm_cli_client.submit_job(
                    job_description, username, jwt_token
    )
else:
   raise HTTPException(
```
you also need to add a check in get_job_metadata

```
if self. slurm_cli_client:
   return await self.slurm_cli_client.get_job_metadata(job_id, username, jwt_token)
else:
   raise HTTPException(
```